ceph/iscsi/gateway.py

```diff
@@ -0,0 +1,183 @@
+"""iSCSI GWCLI Library Module.
+
+This module consists of,
+    - All gwcli operations.
+    - Gateway node access.
+    - gwcli execution methods.
+    - iscsi container related methods.
+"""
+
+from ceph.iscsi.utils import get_iscsi_container
+from cli.utilities.containers import Container
+
+
+class GWCLI:
+    path = "/"
+
+    def __init__(self, node):
+        self.node = node
+        self.target = self.IQNTarget(self)
+        self.disks = self.Disks(self)
+        self.container = Container(self.node)
+        self._iscsi_container = get_iscsi_container(self.node)
+        self.container_id = self.iscsi_container["Names"][0]
+
+    @property
+    def iscsi_container(self):
+        return self._iscsi_container
+
+    @iscsi_container.setter
+    def iscsi_container(self, value):
+        self._iscsi_container = value
+        if not value:
+            self._iscsi_container = get_iscsi_container(self.node)
+
+    @staticmethod
+    def format_command_options(**kwargs):
+        """Construct the iscsi path to define target and its entities.
+
+        Args:
+            kwargs: command options in Dict
+
+        Returns:
+            Str
+        """
+        cmd_opts = str()
+        for k, v in kwargs.items():
+            cmd_opts += f" {k}={v}"
+        return cmd_opts
```

Comment:
IMO, we could move this outside of the class definition. There is `utils` being implemented... it could reside there if you have more implementations.

Reply:
I agree, But This command format is all needed only for GWCLI, and utils could be used for whole iscsi sub-module. But no harm in moving this to utils.py if it make sense.

Reply:
@sunilkumarn417 Any reason for not moving this to utils.py then?

Reply:
@manasagowri This command formatting is w.r.t GWCLI only not generic formatting option.